src/app/context/api/VersionInfo.tsx

```diff
@@ -43,13 +45,22 @@ export function VersionInfoProvider({ children }: PropsWithChildren) {
         firstActivationHeight: 0,
         isError: isHeightError || isVersionError,
         isLoading: isVersionLoading || isHeightLoading,
+        getVersionInfoByHeight: () => undefined,
       };
 
+    const currentVersionInfo = getCurrentGlobalParamsVersion(height + 1, versions);
+
+    const getVersionInfoByHeight = (targetHeight: number) => {
+      const { currentVersion } = getCurrentGlobalParamsVersion(targetHeight, versions);
+      return currentVersion;
+    };
```

Comment:
You don't need to put this logic here. Instead you can extend it by using a custom hook like this one:

```ts
export function useVersionByHeight(height: number) {
  const { versions } = useVersionInfo()
  
  return useMemo(() => getCurrentGlobalParamsVersion(height, versions), [versions, height])
}
```